sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosVectorEmbedding.java

```diff
@@ -0,0 +1,128 @@
+    public CosmosVectorEmbedding() {
+        this.jsonSerializable = new JsonSerializable();
+    }
+
+    /**
+     * Gets the path for the cosmosVectorEmbedding.
+     *
+     * @return path
+     */
+    public String getPath() {
+        return path;
+    }
+
+    /**
+     * Sets the path for the cosmosVectorEmbedding.
+     *
+     * @param path the path for the cosmosVectorEmbedding
+     * @return CosmosVectorEmbedding
+     */
+    public CosmosVectorEmbedding setPath(String path) {
+        if (StringUtils.isEmpty(path)) {
+            throw new NullPointerException("embedding path is empty");
+        }
+
+        if (path.charAt(0) != '/' || path.lastIndexOf('/') != 0) {
+            throw new IllegalArgumentException("");
+        }
+
+        this.path = path;
+        return this;
+    }
+
+    /**
+     * Gets the data type for the cosmosVectorEmbedding.
+     *
+     * @return dataType
+     */
+    public CosmosVectorDataType getDataType() {
+        return CosmosVectorDataType.fromString(dataType);
+    }
+
+    /**
+     * Sets the data type for the cosmosVectorEmbedding.
+     *
+     * @param dataType the data type for the cosmosVectorEmbedding
+     * @return CosmosVectorEmbedding
+     */
+    public CosmosVectorEmbedding setDataType(CosmosVectorDataType dataType) {
+        checkNotNull(dataType, "cosmosVectorDataType cannot be empty");
+        this.dataType = dataType.toString();
+        return this;
+    }
+
+    /**
+     * Gets the dimensions for the cosmosVectorEmbedding.
+     *
+     * @return dimensions
+     */
+    public Long getDimensions() {
+        return dimensions;
+    }
+
+    /**
+     * Sets the dimensions for the cosmosVectorEmbedding.
+     *
+     * @param dimensions the dimensions for the cosmosVectorEmbedding
+     * @return CosmosVectorEmbedding
+     */
+    public CosmosVectorEmbedding setDimensions(Long dimensions) {
+        checkNotNull(dimensions, "dimensions cannot be empty");
+        if (dimensions < 1) {
+            throw new IllegalArgumentException("Dimensions for the embedding has to be a long value greater than 1 " +
+                "for the vector embedding policy");
+        }
+
+        this.dimensions = dimensions;
+        return this;
+    }
+
+    /**
+     * Gets the distanceFunction for the cosmosVectorEmbedding.
+     *
+     * @return distanceFunction
+     */
+    public CosmosVectorDistanceFunction getDistanceFunction() {
+        return CosmosVectorDistanceFunction.fromString(distanceFunction);
+    }
+
+    /**
+     * Sets the distanceFunction for the cosmosVectorEmbedding.
+     *
+     * @param distanceFunction the distanceFunction for the cosmosVectorEmbedding
+     * @return CosmosVectorEmbedding
+     */
+    public CosmosVectorEmbedding setDistanceFunction(CosmosVectorDistanceFunction distanceFunction) {
+        checkNotNull(distanceFunction, "cosmosVectorDistanceFunction cannot be empty");
+        this.distanceFunction = distanceFunction.toString();
```

Comment:
any reason to do toString and then fromString  in the getter?

could  `this.distanceFunction` variable be of type `CosmosVectorDistanceFunction`